src/unstract/sdk/llm.py

```diff
@@ -120,10 +127,75 @@ def _get_max_tokens(self, reserved_for_output: int = 0) -> int:
         """
         return self.MAX_TOKENS - reserved_for_output
 
-    @deprecated("Use the new class LLM")
+    def set_max_tokens(self, max_tokens: int) -> None:
+        """Set the maximum number of tokens that can be used for the LLM.
+
+        Args:
+            max_tokens (int): The number of tokens to be used at the maximum
+
+            Returns:
+                None
+        """
+        self._llm_instance.max_tokens = max_tokens
+
+    def get_class_name(self) -> str:
+        """Gets the class name of the Llama Index LLM.
+
+        Args:
+            NA
+
+            Returns:
+                Class name
+        """
+        return self._llm_instance.class_name()
+
+    @deprecated("Use LLM instead of ToolLLM")
     def get_llm(self, adapter_instance_id: Optional[str] = None) -> LlamaIndexLLM:
+        if not self._llm_instance:
+            self._adapter_instance_id = adapter_instance_id
+            self._initialise()
         return self._llm_instance
 
+    @classmethod
+    @deprecated("Instantiate LLM and call complete() instead")
+    def run_completion(
+        cls,
+        llm: LlamaIndexLLM,
+        platform_api_key: str,
+        prompt: str,
+        retries: int = 3,
+        **kwargs: Any,
+    ) -> Optional[dict[str, Any]]:
+        # Setup callback manager to collect Usage stats
+        CallbackManager.set_callback_manager(
+            platform_api_key=platform_api_key, llm=llm, **kwargs
+        )
+        # Removing specific keys from kwargs
+        new_kwargs = kwargs.copy()
+        for key in [
+            "workflow_id",
+            "execution_id",
+            "adapter_instance_id",
+            "run_id",
+        ]:
+            new_kwargs.pop(key, None)
+
+        try:
+            response: CompletionResponse = llm.complete(prompt, **new_kwargs)
+            match = cls.json_regex.search(response.text)
+            if match:
+                response.text = match.group(0)
+            return {"response": response}
+        # TODO: Handle for all LLM providers
+        except OpenAIAPIError as e:
+            msg = "OpenAI error: "
+            msg += e.message
+            if hasattr(e, "body") and "message" in e.body:
+                msg += e.body["message"]
+            if isinstance(e, OpenAIRateLimitError):
+                raise RateLimitError(msg)
+            raise LLMError(msg) from e
```

Comment:
@gaya3-zipstack Should `e` be surfaced to the end user?

Reply:
@harini-venkataraman over here, we take the user friendly message alone from the exception and pass it. It should ideally get prefixed with context on where the error is coming from - this is handled already I believe. 
The `raise exception from e` will only preserve the traceback for us and will not pass the exception itself to the user